cpp/ql/lib/semmle/code/cpp/ir/dataflow/MustFlow.qll

```diff
@@ -25,18 +24,18 @@ abstract class MustFlowConfiguration extends string {
   /**
    * Holds if `source` is a relevant data flow source.
    */
-  abstract predicate isSource(DataFlow::Node source);
+  abstract predicate isSource(Instruction source);
 
   /**
    * Holds if `sink` is a relevant data flow sink.
    */
-  abstract predicate isSink(DataFlow::Node sink);
+  abstract predicate isSink(Operand sink);
```

Comment:
This looks somewhat asymmetric now. Is this what we want, or should we introduce a `MustFlowNode` that can either be an instruction or an operand (as we also have a `MustFlowPathNode`)?

Reply:
True. I think this is okay since I consider `Instruction`s to be "producers" and `Operand`s to be "consumers", and thus I think it makes sense for an instruction to be a source and an operand to be a sink.

I'd also be totally okay with adding a `MustFlowNode` instead (and having a branch for instructions and a branch for operands).